pkg/utils/error.go

```diff
@@ -0,0 +1,36 @@
+package utils
+
+import (
+	"fmt"
+	"net/http"
+
+	apiErrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// buildApiError returns new error with updated status
```

Comment:
Could you make it clearer? I couldn't tell what the function does only reading this header